File: le-serviceflows/core/src/main/resources/com/latticeengines/serviceflows/core/transforms/std_visidb_ds_namelength.py
#this is the domain length, it is a spam indicator

#this needs to be modified
def metadata():
    return {  'ApprovedUsage'   : 'ModelAndAllInsights'
            , 'Category'        : 'Lead Information'
            , 'FundamentalType' : 'numeric'
            , 'Description'     : 'Name Length'
            , 'DisplayName'     : 'Name Length'
            , 'StatisticalType' : 'ordinal'
            , 'Tags'            : 'Internal' }

def std_visidb_ds_namelength(FirstName,LastName):
    return len(FirstName)+len(LastName)

#function to call
def transform(args, record):
    column1 = args["column1"]
    column2 = args["column2"]
    value1 = record[column1]
    value2 = record[column2]
    return std_visidb_ds_namelength(value1, value2)

File: le-transform/src/main/java/com/latticeengines/transform/v2_0_25/functions/StdVisidbDsNamelength.java
package com.latticeengines.transform.v2_0_25.functions;

import java.util.Map;

import com.latticeengines.transform.exposed.RealTimeTransform;
import com.latticeengines.transform.exposed.metadata.ApprovedUsage;
import com.latticeengines.transform.exposed.metadata.Category;
import com.latticeengines.transform.exposed.metadata.FundamentalType;
import com.latticeengines.transform.exposed.metadata.StatisticalType;
import com.latticeengines.transform.exposed.metadata.Tag;
import com.latticeengines.transform.exposed.metadata.TransformMetadata;

public class StdVisidbDsNamelength implements RealTimeTransform {

    private static final long serialVersionUID = 1437270491579294595L;

    public StdVisidbDsNamelength() {

    }

    public StdVisidbDsNamelength(String modelPath) {

    }

    @Override
    public Object transform(Map<String, Object> arguments, Map<String, Object> record) {
        String column1 = (String) arguments.get("column1");
        String column2 = (String) arguments.get("column2");

        Object firstName = record.get(column1);
        Object lastName = record.get(column2);

        if (firstName == null)
            firstName = "";
        if (lastName == null)
            lastName = "";

        return firstName.toString().length() + lastName.toString().length();
    }

    @Override
    public TransformMetadata getMetadata() {
        TransformMetadata metadata = new TransformMetadata();
        metadata.setApprovedUsage(ApprovedUsage.MODEL_ALLINSIGHTS);
        metadata.setCategory(Category.LEAD_INFORMATION);
        metadata.setFundamentalType(FundamentalType.NUMERIC);
        metadata.setStatisticalType(StatisticalType.ORDINAL);
        metadata.setDescription("Name Length");
        metadata.setDisplayName("Name Length");
        metadata.setTags(Tag.INTERNAL_TRANSFORM);

        return metadata;
    }
}
